rsbinder cannot be built without `async` feature

Trying to add `rsbinder` crate into dependencies without default features results in compilation error:
<details> <summary>Compilation error</summary>
```
error[E0433]: failed to resolve: could not find `async_trait` in the list of imported crates
  --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:59:17
   |
59 |             #[::async_trait::async_trait]
   |                 ^^^^^^^^^^^ could not find `async_trait` in the list of imported crates

error[E0433]: failed to resolve: could not find `async_trait` in the list of imported crates
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:199:17
    |
199 |             #[::async_trait::async_trait]
    |                 ^^^^^^^^^^^ could not find `async_trait` in the list of imported crates

error[E0433]: failed to resolve: could not find `async_trait` in the list of imported crates
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:376:17
    |
376 |             #[::async_trait::async_trait]
    |                 ^^^^^^^^^^^ could not find `async_trait` in the list of imported crates

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:57:117
    |
 57 | ...IBinder, _arg_hasClients: bool) -> crate::BoxFuture<'a, crate::status::Result<()>>;
    |                                              ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0405]: cannot find trait `BinderAsyncRuntime` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:69:31
    |
 69 |                     R: crate::BinderAsyncRuntime + Send + Sync + 'static,
    |                               ^^^^^^^^^^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:41
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                         ^^^^^^^^^^^^^^^^^^

error[E0405]: cannot find trait `BnClientCallbackAdapter` in this scope
  --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:79:32
   |
79 |                     impl<T, R> BnClientCallbackAdapter for Wrapper<T, R>
   |                                ^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0405]: cannot find trait `BinderAsyncRuntime` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:82:35
    |
 82 |                         R: crate::BinderAsyncRuntime + Send + Sync + 'static,
    |                                   ^^^^^^^^^^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:41
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                         ^^^^^^^^^^^^^^^^^^

error[E0405]: cannot find trait `BinderAsyncRuntime` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:94:35
    |
 94 |                         R: crate::BinderAsyncRuntime + Send + Sync + 'static,
    |                                   ^^^^^^^^^^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:41
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                         ^^^^^^^^^^^^^^^^^^

error[E0405]: cannot find trait `BinderAsyncPool` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:144:28
    |
144 |             impl<P: crate::BinderAsyncPool> IClientCallbackAsync<P> for BpClientCallback {
    |                            ^^^^^^^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:24
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                        ^^^^^^^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:145:117
    |
145 | ...IBinder, _arg_hasClients: bool) -> crate::BoxFuture<'a, crate::status::Result<()>> {
    |                                              ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0405]: cannot find trait `BinderAsyncPool` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:159:28
    |
159 |             impl<P: crate::BinderAsyncPool> IClientCallbackAsync<P> for crate::Binder<BnClientCallback>
    |                            ^^^^^^^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:24
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                        ^^^^^^^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:161:117
    |
161 | ...IBinder, _arg_hasClients: bool) -> crate::BoxFuture<'a, crate::status::Result<()>> {
    |                                              ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:197:115
    |
197 | ...rg_binder: &'a crate::SIBinder) -> crate::BoxFuture<'a, crate::status::Result<()>>;
    |                                              ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0405]: cannot find trait `BinderAsyncRuntime` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:209:31
    |
209 |                     R: crate::BinderAsyncRuntime + Send + Sync + 'static,
    |                               ^^^^^^^^^^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:41
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                         ^^^^^^^^^^^^^^^^^^

error[E0405]: cannot find trait `BnServiceCallbackAdapter` in this scope
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:219:32
    |
195 |             pub trait IServiceCallbackAsync<P>: crate::Interface + Send {
    |             ----------------------------------------------------------- similarly named trait `IServiceCallbackAsync` defined here
...
219 |                     impl<T, R> BnServiceCallbackAdapter for Wrapper<T, R>
    |                                ^^^^^^^^^^^^^^^^^^^^^^^^ help: a trait with a similar name exists: `IServiceCallbackAsync`

error[E0405]: cannot find trait `BinderAsyncRuntime` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:222:35
    |
222 |                         R: crate::BinderAsyncRuntime + Send + Sync + 'static,
    |                                   ^^^^^^^^^^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:41
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                         ^^^^^^^^^^^^^^^^^^

error[E0405]: cannot find trait `BinderAsyncRuntime` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:234:35
    |
234 |                         R: crate::BinderAsyncRuntime + Send + Sync + 'static,
    |                                   ^^^^^^^^^^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:41
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                         ^^^^^^^^^^^^^^^^^^

error[E0405]: cannot find trait `BinderAsyncPool` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:284:28
    |
284 |             impl<P: crate::BinderAsyncPool> IServiceCallbackAsync<P> for BpServiceCallback {
    |                            ^^^^^^^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:24
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                        ^^^^^^^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:285:115
    |
285 | ...rg_binder: &'a crate::SIBinder) -> crate::BoxFuture<'a, crate::status::Result<()>> {
    |                                              ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0405]: cannot find trait `BinderAsyncPool` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:299:28
    |
299 |             impl<P: crate::BinderAsyncPool> IServiceCallbackAsync<P> for crate::Binder<BnServiceCallback>
    |                            ^^^^^^^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:24
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                        ^^^^^^^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:301:115
    |
301 | ...rg_binder: &'a crate::SIBinder) -> crate::BoxFuture<'a, crate::status::Result<()>> {
    |                                              ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:359:77
    |
359 | ...   fn r#getService<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Option<crate::SIBinder>>>;
    |                                                                   ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:360:78
    |
360 | ...   fn r#getService2<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<super::Service::Servic...
    |                                                                    ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:361:79
    |
361 | ...   fn r#checkService<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Option<crate::SIBinde...
    |                                                                     ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:362:80
    |
362 | ...   fn r#checkService2<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<super::Service::Serv...
    |                                                                      ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:363:162
    |
363 | ...: bool, _arg_dumpPriority: i32) -> crate::BoxFuture<'a, crate::status::Result<()>>;
    |                                              ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:364:83
    |
364 | ...   fn r#listServices<'a>(&'a self, _arg_dumpPriority: i32) -> crate::BoxFuture<'a, crate::status::Result<Vec<String>>>;
    |                                                                         ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:365:172
    |
365 | ...iceCallback::IServiceCallback>) -> crate::BoxFuture<'a, crate::status::Result<()>>;
    |                                              ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:366:174
    |
366 | ...iceCallback::IServiceCallback>) -> crate::BoxFuture<'a, crate::status::Result<()>>;
    |                                              ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:367:77
    |
367 |                 fn r#isDeclared<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<bool>>;
    |                                                                             ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:368:88
    |
368 | ...   fn r#getDeclaredInstances<'a>(&'a self, _arg_iface: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Vec<String>>>;
    |                                                                              ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:369:83
    |
369 | ...   fn r#updatableViaApex<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Option<String>>>;
    |                                                                         ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:370:88
    |
370 | ...   fn r#getUpdatableNames<'a>(&'a self, _arg_apexName: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Vec<String>>>;
    |                                                                              ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:371:84
    |
371 | ...   fn r#getConnectionInfo<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Option<super::Co...
    |                                                                          ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:372:203
    |
372 | ...ientCallback::IClientCallback>) -> crate::BoxFuture<'a, crate::status::Result<()>>;
    |                                              ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:373:122
    |
373 | ...g_service: &'a crate::SIBinder) -> crate::BoxFuture<'a, crate::status::Result<()>>;
    |                                              ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:374:66
    |
374 | ...   fn r#getServiceDebugInfo<'a>(&'a self) -> crate::BoxFuture<'a, crate::status::Result<Vec<super::ServiceDebugInfo::Servi...
    |                                                        ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0405]: cannot find trait `BinderAsyncRuntime` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:401:31
    |
401 |                     R: crate::BinderAsyncRuntime + Send + Sync + 'static,
    |                               ^^^^^^^^^^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:41
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                         ^^^^^^^^^^^^^^^^^^

error[E0405]: cannot find trait `BnServiceManagerAdapter` in this scope
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:411:32
    |
411 |                     impl<T, R> BnServiceManagerAdapter for Wrapper<T, R>
    |                                ^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0405]: cannot find trait `BinderAsyncRuntime` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:414:35
    |
414 |                         R: crate::BinderAsyncRuntime + Send + Sync + 'static,
    |                                   ^^^^^^^^^^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:41
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                         ^^^^^^^^^^^^^^^^^^

error[E0405]: cannot find trait `BinderAsyncRuntime` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:426:35
    |
426 |                         R: crate::BinderAsyncRuntime + Send + Sync + 'static,
    |                                   ^^^^^^^^^^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:41
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                         ^^^^^^^^^^^^^^^^^^

error[E0405]: cannot find trait `BinderAsyncPool` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:921:28
    |
921 |             impl<P: crate::BinderAsyncPool> IServiceManagerAsync<P> for BpServiceManager {
    |                            ^^^^^^^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:24
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                        ^^^^^^^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:922:77
    |
922 | ...   fn r#getService<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Option<crate::SIBinder>...
    |                                                                   ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:935:78
    |
935 | ...   fn r#getService2<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<super::Service::Servic...
    |                                                                    ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:948:79
    |
948 | ...   fn r#checkService<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Option<crate::SIBinde...
    |                                                                     ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:961:80
    |
961 | ...   fn r#checkService2<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<super::Service::Serv...
    |                                                                      ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:974:162
    |
974 | ...: bool, _arg_dumpPriority: i32) -> crate::BoxFuture<'a, crate::status::Result<()>> {
    |                                              ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
   --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:987:83
    |
987 | ...   fn r#listServices<'a>(&'a self, _arg_dumpPriority: i32) -> crate::BoxFuture<'a, crate::status::Result<Vec<String>>> {
    |                                                                         ^^^^^^^^^ not found in the crate root
    |
note: found an item that was configured out
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
    |
165 | #[cfg(feature = "async")]
    |       ----------------- the item is gated behind the `r#async` feature
166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
    |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1000:172
     |
1000 | ...iceCallback::IServiceCallback>) -> crate::BoxFuture<'a, crate::status::Result<()>> {
     |                                              ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1013:174
     |
1013 | ...iceCallback::IServiceCallback>) -> crate::BoxFuture<'a, crate::status::Result<()>> {
     |                                              ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1026:77
     |
1026 |                 fn r#isDeclared<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<bool>> {
     |                                                                             ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1039:88
     |
1039 | ...   fn r#getDeclaredInstances<'a>(&'a self, _arg_iface: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Vec<String>...
     |                                                                              ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1052:83
     |
1052 | ...   fn r#updatableViaApex<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Option<String>>> {
     |                                                                         ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1065:88
     |
1065 | ...   fn r#getUpdatableNames<'a>(&'a self, _arg_apexName: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Vec<String>...
     |                                                                              ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1078:84
     |
1078 | ...   fn r#getConnectionInfo<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Option<super::C...
     |                                                                          ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1091:203
     |
1091 | ...ientCallback::IClientCallback>) -> crate::BoxFuture<'a, crate::status::Result<()>> {
     |                                              ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1104:122
     |
1104 | ...g_service: &'a crate::SIBinder) -> crate::BoxFuture<'a, crate::status::Result<()>> {
     |                                              ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1117:66
     |
1117 | ...   fn r#getServiceDebugInfo<'a>(&'a self) -> crate::BoxFuture<'a, crate::status::Result<Vec<super::ServiceDebugInfo::Serv...
     |                                                        ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0405]: cannot find trait `BinderAsyncPool` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1131:28
     |
1131 |             impl<P: crate::BinderAsyncPool> IServiceManagerAsync<P> for crate::Binder<BnServiceManager>
     |                            ^^^^^^^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:24
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                        ^^^^^^^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1133:77
     |
1133 | ...   fn r#getService<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Option<crate::SIBinder...
     |                                                                   ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1136:78
     |
1136 | ...   fn r#getService2<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<super::Service::Servi...
     |                                                                    ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1139:79
     |
1139 | ...   fn r#checkService<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Option<crate::SIBind...
     |                                                                     ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1142:80
     |
1142 | ...   fn r#checkService2<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<super::Service::Ser...
     |                                                                      ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1145:162
     |
1145 | ...: bool, _arg_dumpPriority: i32) -> crate::BoxFuture<'a, crate::status::Result<()>> {
     |                                              ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1148:83
     |
1148 | ...   fn r#listServices<'a>(&'a self, _arg_dumpPriority: i32) -> crate::BoxFuture<'a, crate::status::Result<Vec<String>>> {
     |                                                                         ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1151:172
     |
1151 | ...iceCallback::IServiceCallback>) -> crate::BoxFuture<'a, crate::status::Result<()>> {
     |                                              ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1154:174
     |
1154 | ...iceCallback::IServiceCallback>) -> crate::BoxFuture<'a, crate::status::Result<()>> {
     |                                              ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1157:77
     |
1157 |                 fn r#isDeclared<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<bool>> {
     |                                                                             ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1160:88
     |
1160 | ...   fn r#getDeclaredInstances<'a>(&'a self, _arg_iface: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Vec<String>...
     |                                                                              ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1163:83
     |
1163 | ...   fn r#updatableViaApex<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Option<String>>> {
     |                                                                         ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1166:88
     |
1166 | ...   fn r#getUpdatableNames<'a>(&'a self, _arg_apexName: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Vec<String>...
     |                                                                              ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1169:84
     |
1169 | ...   fn r#getConnectionInfo<'a>(&'a self, _arg_name: &'a str) -> crate::BoxFuture<'a, crate::status::Result<Option<super::C...
     |                                                                          ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1172:203
     |
1172 | ...ientCallback::IClientCallback>) -> crate::BoxFuture<'a, crate::status::Result<()>> {
     |                                              ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1175:122
     |
1175 | ...g_service: &'a crate::SIBinder) -> crate::BoxFuture<'a, crate::status::Result<()>> {
     |                                              ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

error[E0412]: cannot find type `BoxFuture` in the crate root
    --> /home/user/MyProject/target/debug/build/rsbinder-3fb6b667c3d590e8/out/service_manager_16.rs:1178:66
     |
1178 | ...   fn r#getServiceDebugInfo<'a>(&'a self) -> crate::BoxFuture<'a, crate::status::Result<Vec<super::ServiceDebugInfo::Serv...
     |                                                        ^^^^^^^^^ not found in the crate root
     |
note: found an item that was configured out
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rsbinder-0.4.3/src/lib.rs:166:61
     |
 165 | #[cfg(feature = "async")]
     |       ----------------- the item is gated behind the `r#async` feature
 166 | pub use binder_async::{BinderAsyncPool, BinderAsyncRuntime, BoxFuture};
     |                                                             ^^^^^^^^^

Some errors have detailed explanations: E0405, E0412, E0433.
For more information about an error, try `rustc --explain E0405`.
error: could not compile `rsbinder` (lib) due to 75 previous errors
```
</details>

--- a/example-hello/Cargo.toml
+++ b/example-hello/Cargo.toml
@@ -6,10 +6,6 @@ edition = { workspace = true }
 license = { workspace = true }
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
-# [features]
-# default = ["async"]
-# sync = ["rsbinder/sync", "rsbinder-aidl/sync"]
-# async = ["rsbinder/async", "rsbinder-aidl/async"]
 
 [dependencies]
 rsbinder = { workspace = true, features = ["android_11_plus"] }
@@ -17,4 +13,4 @@ env_logger = { workspace = true }
 async-trait = { workspace = true }
 
 [build-dependencies]
-rsbinder-aidl = { workspace = true }
+rsbinder-aidl = { workspace = true, features = ["async"] }

--- a/rsbinder-aidl/Cargo.toml
+++ b/rsbinder-aidl/Cargo.toml
@@ -12,8 +12,7 @@ keywords.workspace = true
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 [features]
-default = ["async"]
-sync = []
+default = []
 async = []
 
 [dependencies]

--- a/rsbinder-aidl/README.md
+++ b/rsbinder-aidl/README.md
@@ -2,22 +2,38 @@
 This is an AIDL compiler for **rsbinder**.
 
 ## How to use the AIDL Code Generator
-* Add the build-dependencies to Cargo.toml:
-```
+Add dependencies to Cargo.toml:
+```toml
+[dependencies]
+rsbinder = "0.5"
+
 [build-dependencies]
-rsbinder-aidl = "0.4.0"
+rsbinder-aidl = { version = "0.5", features = ["async"] }
 ```
-* Create a build.rs file in the root folder of the crate.
-* Add use std::path::PathBuf; to build.rs.
-* Add the following content:
+
+Create a build.rs file:
+```rust
+use std::path::PathBuf;
+
+fn main() {
+    rsbinder_aidl::Builder::new()
+        .source(PathBuf::from("aidl/IMyService.aidl"))
+        .output(PathBuf::from("my_service.rs"))
+        .generate()
+        .unwrap();
+}
 ```
-rsbinder_aidl::Builder::new()
-    .source(PathBuf::from("aidl/....")
-    .source(PathBuf::from("aidl/....")
-    .source(PathBuf::from("aidl/....")
-    .output(PathBuf::from("aidl_name.rs")
-    .generate().unwrap()
+
+### Sync-only Setup
+For environments without async runtime:
+```toml
+[dependencies]
+rsbinder = { version = "0.5", default-features = false }
+
+[build-dependencies]
+rsbinder-aidl = "0.5"
 ```
+
 ## How to create AIDL file
 Please read Android AIDL documents.
 

--- a/rsbinder/Cargo.toml
+++ b/rsbinder/Cargo.toml
@@ -14,7 +14,6 @@ keywords.workspace = true
 
 [features]
 default = ["tokio"]
-sync = ["rsbinder-aidl/sync"]
 tokio = ["async", "tokio/full"]
 async = ["rsbinder-aidl/async", "async-trait"]
 android_11 = []
@@ -38,7 +37,7 @@ tokio = { workspace = true, optional = true }
 rsproperties.workspace = true
 
 [build-dependencies]
-rsbinder-aidl = { workspace = true }
+rsbinder-aidl = { workspace = true, features = ["async"] }
 
 [dev-dependencies]
 env_logger = { workspace = true }

--- a/tests/Cargo.toml
+++ b/tests/Cargo.toml
@@ -16,4 +16,4 @@ async-trait = { workspace = true }
 log.workspace = true
 
 [build-dependencies]
-rsbinder-aidl = { workspace = true }
+rsbinder-aidl = { workspace = true, features = ["async"] }

